cpp/src/arrow/flight/sql/odbc/odbcabstraction/calendar_utils.cc

```diff
@@ -17,6 +17,7 @@
 
 #include "odbcabstraction/calendar_utils.h"
 
+#include <boost/date_time/posix_time/posix_time.hpp>
```

Comment:
Can we use `std::chrono` instead of Boost? (Currently, Arrow C++ requires C++17 but we'll require C++20 soon https://github.com/apache/arrow/issues/44792 . So we can use C++20 in ODBC support.)
In general, we want to avoid using Boost as much as possible. It's difficult to maintain.